src/messagequeue/adapters/eventhub.ts

```diff
@@ -23,8 +23,42 @@ export const create = (
     await producer.close();
   };
 
+  const sendMessagesBatch = async(messages: Message[], batchOptions: CreateBatchOptions) => {
+    try {
+      const producer: EventHubProducerClient = new EventHubProducerClient(
+        connectionString,
+        name
+      );
+
+      let batch = await producer.createBatch(batchOptions);
+      let batchesSent = 0
+
+      for (let i = 0; i < messages.length; i++ )
+      {
+        const messageEvent: EventData = {
+          body: messages[i],
+        };
+        if (!batch.tryAdd(messageEvent) || i === messages.length-1) {
+          console.log(
+            `Sending ${batch.count} messages as a single batch. Current Batch Number ${batchesSent + 1}.`
+          );
+          await producer.sendBatch(batch);
+          batchesSent++;
+
+          // create a new batch to house the next set of messages
+          batch =  await producer.createBatch(batchOptions);
+        }
+      }
+      await producer.close();
+      console.log(`Batch Messaging Complete, ${batchesSent} batches sent.`);
+    } catch (err) {
+      console.error("Error when creating & sending a batch of messages: ", err);
```

Comment:
`throw new Error(<message>)` will actually halt execution here if something goes wrong whereas "console.error(<message>)` will not.

Reply:
Agreed, don't try/catch a whole function just to not throw the error.
Let the caller of this function decide whether they want it to error or not.

Reply:
Now throwing error